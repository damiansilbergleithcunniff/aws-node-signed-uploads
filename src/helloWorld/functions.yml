handler: src/helloWorld/handler.default
name: ${self:provider.stage}-${self:service}-helloWorld
memorySize: 128
events:
  - http:
      path: helloWorld
      method: get
      cors: true
