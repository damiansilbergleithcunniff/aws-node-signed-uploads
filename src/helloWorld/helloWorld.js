const helloWorld = (status, msg, event) => {
  return {
    statusCode: status,
    body: JSON.stringify({
      message: msg,
      input: event,
    }),
  };
};

export default helloWorld;
