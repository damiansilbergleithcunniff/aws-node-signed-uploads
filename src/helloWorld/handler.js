import helloWorld from './helloWorld';

const main = (event, context, callback) => {
  const response = helloWorld(200, 'You should rework this component', event);

  callback(null, response);
};

export default main;
