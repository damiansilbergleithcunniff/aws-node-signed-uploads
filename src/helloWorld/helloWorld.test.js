import helloWorld from './helloWorld';

describe('helloWorld', () => {
  test('creates correct response', () => {
    const dummyStatus = 'DUMMY STATUS';
    const dummyMessage = 'DUMMY MESSAGE';
    const dummyEvent = 'DUMMY EVENT';
    const response = helloWorld(dummyStatus, dummyMessage, dummyEvent);

    expect(response).toMatchSnapshot();
  });
});
