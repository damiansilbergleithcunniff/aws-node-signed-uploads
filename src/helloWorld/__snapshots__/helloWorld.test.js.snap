// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helloWorld creates correct response 1`] = `
Object {
  "body": "{\\"message\\":\\"DUMMY MESSAGE\\",\\"input\\":\\"DUMMY EVENT\\"}",
  "statusCode": "DUMMY STATUS",
}
`;
